### PR TITLE
nlplug-findfs: test script

### PR DESCRIPTION
**Pull request by 7heo**

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	eecho "  help (this help)"
+	eecho
+	eecho "flags:"
+	eecho "  -h: help (this help)"
+	eecho "  -d: turn on nlplug-findfs debug output"
+	eecho "  -y: skip confirmation at program start"
+	eecho "  -x: clean all (including produced binaries)"
+	eecho
+	exit 1
+}
```
Comment (jirutka) [inline test.sh:29]: Please replace this noise with simple heredoc:

``` sh
usage () {
    cat <<-EOF
        $0 [flags] [operation]

        operation:
          full (default)
          clean
          help (this help)

        flags:
          -h: help (this help)
          -d: turn on nlplug-findfs debug output
          -y: skip confirmation at program start
          -x: clean all (including produced binaries)
    EOF
    exit 1
}
```

File: test.sh [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+		"-x") clean_all=1; shift;;
+		"-h"|"--help") usage; shift;;
+		"-"*) shift;; # Ignore erroneous flags
+		*) break;;
+	esac
+done
+
+[ $# -eq 1 ] && operation=$1
+
+[ $noconfirm -eq 0 ] && echo "Warning: this script is going to de-setup all your existing /dev/loop* devices at exit. Press enter to continue, or ^C to cancel."
```
Comment (jirutka) [inline test.sh]: This should be printed on stderr. Also please wrap to 80 chars.

**Review by jirutka — CHANGES_REQUESTED** [review]

File: test.sh (hunk 1 above)
Comment (7heo) [reply test.sh:29]: `./test.sh: line 110: syntax error: unexpected end of file (expecting "}")`

This is the exact reason why I didn't go with the heredocs in the first place: it prevents the use of indentation, as always with heredocs.

**Comment by jirutka** [comment]
Do you use tabs for indentation? `<<-` works only with tabs.

File: test.sh (hunk 1 above)
Comment (7heo) [reply test.sh:29]: @jirutka I did not know that. With tabs it works fine indeed.

Comment (jirutka) [reply test.sh:29]: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_07_04:

> If the redirection symbol is "<<-", all leading <tab> characters shall be stripped from input lines and the line containing the trailing delimiter.

That’s the reason why I use tabs in shell scripts, although I prefer spaces evewyhere else.

Comment (7heo) [reply test.sh:29]: Thank you for showing me that. :)

Comment (jirutka) [reply test.sh:29]: You’re welcome. ;)

**Pull request merged**